cura/Machines/MachineNode.py

```diff
@@ -0,0 +1,172 @@
+# Copyright (c) 2019 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+from typing import Dict, List
+
+from UM.Logger import Logger
+from UM.Signal import Signal
+from UM.Util import parseBool
+from UM.Settings.ContainerRegistry import ContainerRegistry  # To find all the variants for this machine.
+from cura.Machines.ContainerNode import ContainerNode
+from cura.Machines.QualityChangesGroup import QualityChangesGroup  # To construct groups of quality changes profiles that belong together.
+from cura.Machines.QualityGroup import QualityGroup  # To construct groups of quality profiles that belong together.
+from cura.Machines.QualityNode import QualityNode
+from cura.Machines.VariantNode import VariantNode
+
+##  This class represents a machine in the container tree.
+#
+#   The subnodes of these nodes are variants.
+class MachineNode(ContainerNode):
+    def __init__(self, container_id: str) -> None:
```

Comment:
@Ghostkeeper explained to me that `MachineNode`s are identified/organized with `definition_id`s. So, maybe rename `container_id`s or add documentation here to clarify this?

Reply:
The ContainerNodes still store their own `container_id` regardless of how they are stored in the parent nodes. In the parent node of this node (ContainerTree) it is documented that machines are stored by their definition ID.